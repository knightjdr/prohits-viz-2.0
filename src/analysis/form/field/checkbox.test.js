import React from 'react';
import { mount } from 'enzyme';

import Checkbox from './checkbox';
import TestForm from './__mocks__/form-wrapper';

const inputChange = jest.fn();
const onChange = jest.fn();

describe('Checkbox', () => {
  test('Renders with false as initial value', () => {
    const wrapper = mount(
      <TestForm
        input={{
          change: inputChange,
          value: undefined,
        }}
        meta={{}}
      >
        <Checkbox
          input={{}}
          name="TestCheckbox"
          onChange={onChange}
          placeHolder="Checkbox"
          style={{}}
        />
      </TestForm>,
    );
    expect(wrapper).toMatchSnapshot();
    const input = wrapper.find('input');
    expect(input.props().checked).toBeFalsy();
  });

  test('On change called on checkbox click', () => {
    const wrapper = mount(
      <TestForm
        input={{
          change: inputChange,
          value: undefined,
        }}
        meta={{}}
      >
        <Checkbox
          input={{}}
          name="TestCheckbox"
          onChange={onChange}
          placeHolder="Checkbox"
          style={{}}
        />
      </TestForm>,
    );
    const input = wrapper.find('input');
    input.simulate('change');
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  test('Can be set via input.value', () => {
    const wrapper = mount(
      <TestForm
        input={{
          change: inputChange,
          value: undefined,
        }}
        meta={{}}
      >
        <Checkbox
          input={{}}
          name="TestCheckbox"
          onChange={onChange}
          placeHolder="Checkbox"
          style={{}}
        />
      </TestForm>,
    );
    wrapper.setProps({
      input: {
        change: inputChange,
        value: true,
      },
    });
    const input = wrapper.find('input');
    expect(input.props().checked).toBeTruthy();
  });

  test('Submit called on button click and submits with no errors', () => {
    const onSubmitSpy = jest.fn();
    const wrapper = mount(
      <TestForm
        input={{
          change: inputChange,
          value: undefined,
        }}
        meta={{}}
        onSubmit={onSubmitSpy}
      >
        <Checkbox
          input={{}}
          name="TestCheckbox"
          onChange={onChange}
          placeHolder="Checkbox"
          style={{}}
        />
      </TestForm>,
    );
    const button = wrapper.find('button');
    button.simulate('submit');
    expect(onSubmitSpy).toHaveBeenCalledTimes(1);
    const input = wrapper.find('input');
    expect(input.props().checked).toBeFalsy();
  });

  test('Can add custom style', () => {
    const wrapper = mount(
      <TestForm
        input={{
          change: inputChange,
          value: undefined,
        }}
        meta={{}}
      >
        <Checkbox
          input={{}}
          name="TestCheckbox"
          onChange={onChange}
          placeHolder="Checkbox"
          style={{ backgroundColor: '#000' }}
        />
      </TestForm>,
    );
    const checkboxStyle = wrapper.find('Checkbox').first().props().style;
    expect(checkboxStyle).toHaveProperty('backgroundColor', '#000');
  });
});
