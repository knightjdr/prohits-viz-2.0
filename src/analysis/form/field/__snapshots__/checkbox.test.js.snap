// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Renders with false as initial value 1`] = `
<TestForm
  input={
    Object {
      "change": [Function],
      "value": undefined,
    }
  }
  meta={Object {}}
  onSubmit={[Function]}
>
  <Form
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
    prefixCls="ant-form"
  >
    <form
      className="ant-form ant-form-horizontal"
      onSubmit={[Function]}
    >
      <CustomCheckbox
        input={
          Object {
            "change": [Function],
            "value": undefined,
          }
        }
        label={null}
        meta={Object {}}
        name="TestCheckbox"
        onChange={[Function]}
        placeHolder="Checkbox"
        style={Object {}}
      >
        <FormItem
          colon={true}
          hasFeedback={false}
          label={null}
          prefixCls="ant-form"
        >
          <Row
            className="ant-form-item"
            gutter={0}
          >
            <div
              className="ant-row ant-form-item"
            >
              <Col
                className="ant-form-item-control-wrapper"
                key=".$wrapper"
              >
                <div
                  className="ant-form-item-control-wrapper"
                >
                  <div
                    className="ant-form-item-control"
                  >
                    <span
                      className="ant-form-item-children"
                    >
                      <Checkbox
                        checked={false}
                        indeterminate={false}
                        onChange={[Function]}
                        prefixCls="ant-checkbox"
                        style={Object {}}
                      >
                        <label
                          className="ant-checkbox-wrapper"
                          style={Object {}}
                        >
                          <Checkbox
                            checked={false}
                            className=""
                            defaultChecked={false}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            prefixCls="ant-checkbox"
                            style={Object {}}
                            type="checkbox"
                          >
                            <span
                              className="ant-checkbox"
                              style={Object {}}
                            >
                              <input
                                checked={false}
                                className="ant-checkbox-input"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                type="checkbox"
                              />
                              <span
                                className="ant-checkbox-inner"
                              />
                            </span>
                          </Checkbox>
                        </label>
                      </Checkbox>
                    </span>
                    <Animate
                      animation={Object {}}
                      component=""
                      componentProps={Object {}}
                      key="help"
                      onAppear={[Function]}
                      onEnd={[Function]}
                      onEnter={[Function]}
                      onLeave={[Function]}
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="show-help"
                    />
                  </div>
                </div>
              </Col>
            </div>
          </Row>
        </FormItem>
      </CustomCheckbox>
      <button
        type="submit"
      />
    </form>
  </Form>
</TestForm>
`;
